src/Analyzers/CSharp/Analyzers/UseCollectionExpression/AbstractCSharpUseCollectionExpressionDiagnosticAnalyzer.cs

```diff
@@ -26,14 +25,14 @@ internal abstract class AbstractCSharpUseCollectionExpressionDiagnosticAnalyzer
     protected readonly DiagnosticDescriptor UnnecessaryCodeDescriptor;
 
     protected AbstractCSharpUseCollectionExpressionDiagnosticAnalyzer(string diagnosticId, EnforceOnBuild enforceOnBuild)
-        : base(ImmutableDictionary<DiagnosticDescriptor, IOption2>.Empty
-            // Ugly hack.  We need to create a descriptor to pass to our base *and* assign to one of our fields.
-            // The conditional pattern form lets us do that.
-            .Add(CreateDescriptor(diagnosticId, enforceOnBuild, isUnnecessary: false) is var descriptor ? descriptor : null, CodeStyleOptions2.PreferCollectionExpression)
-            .Add(CreateDescriptor(diagnosticId, enforceOnBuild, isUnnecessary: true) is var unnecessaryCodeDescriptor ? unnecessaryCodeDescriptor : null, CodeStyleOptions2.PreferCollectionExpression))
+        : base(
+            [
+                (CreateDescriptor(diagnosticId, enforceOnBuild, isUnnecessary: false), CodeStyleOptions2.PreferCollectionExpression),
+                (CreateDescriptor(diagnosticId, enforceOnBuild, isUnnecessary: true), CodeStyleOptions2.PreferCollectionExpression)
```

Comment:
>(CreateDescriptor(diagnosticId, enforceOnBuild, isUnnecessary: true), CodeStyleOptions2.PreferCollectionExpression) [](http://example.com/codeflow?start=16&length=115)

@CyrusNajmabadi -- Do you know the expected behavior here? Was this code intentional in only adding a single item to the immutable dictionary previously?

This is causing a test failure (UseCollectionExpressionForArrayTests.TestTargetTypedInitializer5) because it's VerifyCS().Diagnostic call wants there to only be a single diagnostic descriptor for this class. I'm not sure whether the change the test, or make it so this code does indeed only add a single diagnostic descriptor for this analyzer.

Reply:
i do not understand what any of this code did :)